dissect/target/tools/query.py

```diff
@@ -57,7 +83,16 @@ def main():
     parser.add_argument("-f", "--function", help="function to execute")
     parser.add_argument("--child", help="load a specific child path or index")
     parser.add_argument("--children", action="store_true", help="include children")
-    parser.add_argument("-l", "--list", action="store_true", help="list available plugins and loaders")
+    parser.add_argument(
+        "-l",
+        "--list",
+        action="store",
+        nargs="?",
+        const="*",
+        default=False,
+        help="list (matching) available plugins and loaders",
+    )
+    parser.add_argument("--compatible", action="store_true", help="full plugin compatibility check")
```

Comment:
Can you sell me on this addition? I still fail to see the added benefit of this flag. I can see it being useful when being used exclusively in combination with `-l`, but not for execution, because it will only add latency for no real benefit.

Reply:
- Clean output, no unnecessary notices between the records when using with a wildcard
- Uniform output, i.e. same records for every target when used with a wildcard, this is nice for streaming to csvs or databases (this feature was rejected so this no longer applies)

I believe @pyrco also mentioned that this was a cleaner way of doing things.
To be honest, I thought the 2nd reason was quite nice, but now that no longer applies I do not see the benefit anymore, however just to be sure @pyrco are you okay with removing this flag?

Reply:
I personally don't find the added complexity and latency worth it for reason 1.

Reply:
I had in mind that `--compatible` was implicit once you specified a target with `-l`, but apparently it isn't.
I think it is useful to have `--compatible` in combination with `-l`, I think it is even more useful to have `--compatible` be automatic with `-l` if you also supply a target.

From the looks of it only implementing `--compatible` behavior for the `-l` option seems to be doable with much less complexity. I think weighing this against the "correctness" of not executing things that you know beforehand won't work does slightly tip in favor of a less complex implementation.

Reply:
How does that work if you specify multiple targets by the way? Do you just print the compatible plugin list for each specified target?

Reply:
No just the plugins that match any target.

Reply:
Adjusted, --compatible is now implied with list only

Reply:
The seeming complexity in `query.py` stayed though, is that intentional?

Reply:
I will see if I can simplify it further